index.js

```diff
@@ -186,6 +190,28 @@ HomeAssistantPlatform.prototype = {
           accessory = new HomeAssistantSwitch(that.log, entity, that, 'input_boolean')
         }else if (entity_type == 'fan'){
           accessory = new HomeAssistantFan(that.log, entity, that)
+        }else if (entity_type == 'binary_sensor' && entity.attributes && entity.attributes.sensor_class) {
+          switch(entity.attributes.sensor_class) {
+            case 'opening':
+              var opening_type;
+              if (entity.attributes.homebridge_opening_type && entity.attributes.homebridge_opening_type == 'window') {
+                opening_type = Service.Window;
+              }else{
+                opening_type = Service.Door;
+              }
```

Comment:
I thought this was actually in the sensor, heh.I think the runtime decisions of characteristics should probably be in `binary_sensor.js`.